src/base/profile.h

```diff
@@ -70,16 +70,15 @@ class Profile
     static const Profile &instance();
 
 private:
-    Profile(Private::Profile *impl, Private::PathConverter *pathConverter);
+    Profile(const QString &rootProfilePath, const QString &configurationName, bool convertPathsToProfileRelative);
     ~Profile();
 
     friend class ::Application;
-    static void initialize(const QString &rootProfilePath, const QString &configurationName,
-                                             bool convertPathsToProfileRelative);
+    static void initialize(const QString &rootProfilePath, const QString &configurationName, bool convertPathsToProfileRelative);
     void ensureDirectoryExists(SpecialFolder folder);
 
-    const std::unique_ptr<Private::Profile> m_profileImpl;
-    const std::unique_ptr<Private::PathConverter> m_pathConverterImpl;
+    const Private::Profile *m_profileImpl;
+    const Private::PathConverter *m_pathConverterImpl;
```

Comment:
Why did you change these two declarations? Please keep them as "std::unique_ptr".

Reply:
Because I wasn't able to compile with it. If you are able to compile then please include changes you made.

Reply:
Seems you created memory leaks now.

Reply:
>Because I wasn't able to compile with it.

Seems you just need to remove "const" to make it compiled.

Reply:
Looks like removing `const` doesn't help. Anything else I should try?

Reply:
Sorry, I can't understand why some problem is here. It was "std::unique_ptr" before and it was compiled w/o errors...
Please provide corresponding error messages.

Reply:
@Chocobo1 any ideas how this piece of code should be changed?

Reply:
@glassez here are the error messages from Qt Creator in windows: https://pastebin.com/dH5a5cDk
and here are errors from build in Ubuntu: https://pastebin.com/pzpvMYyH
and without const: https://pastebin.com/3epiF2YQ and https://pastebin.com/VSTLShT8

Can't you just try to compile it yourself and check the errors?

Reply:
```c++
if (rootProfilePath.isEmpty())
        m_profileImpl = std::make_unique<Private::DefaultProfile>(configurationName);
    else
        m_profileImpl = std::make_unique<Private::CustomProfile>(rootProfilePath, configurationName);


    ensureDirectoryExists(SpecialFolder::Cache);
    ensureDirectoryExists(SpecialFolder::Config);
    ensureDirectoryExists(SpecialFolder::Data);

    if (convertPathsToProfileRelative)
        m_pathConverterImpl = std::make_unique<Private::Converter>(m_profileImpl->baseDirectory());
    else
        m_pathConverterImpl = std::make_unique<Private::NoConvertConverter>();
```

Reply:
@glassez Thanks a lot for fixing it!
Let me know please if I should still change something.